src/app/login/login.component.ts

```diff
@@ -0,0 +1,61 @@
+import {Component, OnInit} from '@angular/core';
+import {FormGroup, FormControl} from '@angular/forms';
+import {User} from '../user';
+import {LoginService} from '../login.service';
+import {Router} from '@angular/router';
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent implements OnInit {
+  public nestedForm: FormGroup = new FormGroup({
+    emailForm: new FormControl(''),
+    passwordForm: new FormControl(''),
+  });
+
+  constructor(private loginService: LoginService, private router: Router) {}
+
+  ngOnInit(): void {}
+
+  public login(email: string, password: string): void {
+    /* const email = this.nestedForm.value.emailForm.email;
+    const password = this.nestedForm.value.passwordForm.pwd; */
+
+    console.log('mail: ', email);
+    console.log('pass: ', password);
+    /* console.log('mail-regexp: ', regexMail.test(email));
+    console.log('pass-regexp: ', regexPwd.test(password)); */
+    /* if (email === '' || typeof email === 'undefined') {
+      console.log('email vuota');
+    }
+    if (password === '' || typeof password === 'undefined') {
+      console.log('password vuota');
+    } */
+    // console.log('################################################');
+    if (!this.validateInput(email, password)) {
+      return;
+    }
+    const user = new User(email, password);
+    this.loginService.login(user).subscribe((response: Response) => {
+      console.log(response);
+      if (response.status === 200) {
+        this.router.navigate(['/home']);
+      }
+    });
+  }
+  public validateInput(email: string, password: string): boolean {
+    const regexMail = new RegExp(
+      '^([a-zA-Z0-9_.+-])+@(([a-zA-Z0-9-])+.)+([a-zA-Z0-9])+$',
+    );
+    const regexPwd = new RegExp(
+      // Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character
+      '^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$',
+    );
```

Comment:
queste regex scritte esplicitamente potrebbero dar problemi; se volessimo far le cose bene bisognerebbe fare un endpoint specifico e una query all'API in background appena viene caricata la pagina che ritorna le regex di validazione, perché anche se la verifica è client side è il server a "comandare". se no se cambiamo i criteri dentro al server bisogna trovare tutti i validatori e cambiarli

Reply:
aggiungo che le regex sono diverse da quelle delle API

Reply:
Si quelle della API non funzionavano, nel mio server stoplight avevamo provato a cambiarle con quelle di @fabioscettro e funzionavano

Reply:
Allora bisogna cambiare quelle delle API, ad esempio secondo le API la password non può contenere dei numeri

Reply:
Allora bisogna cambiare quelle delle API, ad esempio secondo le API la password non può contenere dei numeri

Reply:
lo faccio adesso che lavoro sul server così almeno funziona. mettete un fixme nella web app per trovare un modo più elegante di gestire la cosa